GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.Designer.cs

```diff
@@ -597,5 +634,7 @@ private void InitializeComponent()
         private System.Windows.Forms.TableLayoutPanel tableLayoutPanel5;
         private System.Windows.Forms.ComboBox ProtocolDropdownList;
         private System.Windows.Forms.Label ProtocolLabel;
+        private System.Windows.Forms.CheckBox depthCb;
+        private System.Windows.Forms.TextBox depthTb;
```

Comment:
This should be represented as a numeric up/down control with sensible limits.

Reply:
What would those limits be? I couldn't find any restrictions in Git documentation.

Reply:
It could be just >=1 (or 0==disabled).
Default 1 if enabled?

A up/down is nicer than parsing text - but if you want to use this option, you probably know what you are doing anyway

Reply:
> (or 0==disabled).

👍 For some reason it didn't occur to me. This way we don't need the checkbox.
The label could be "Depth limit (0 - no limit)".

I supposed max can be set at 100, if you need deeper you may as well clone the whole thing.
Thoughts?

Reply:
> I supposed max can be set at 100, if you need deeper you may as well clone the whole thing.

I see no reason enforce a limit

Reply:
We need to set the upper limit on the updown control. 
Obviously after some point (e.g. depth 100 or depth 1,000) specifying a depth is essentially the same as doing a full clone. We just need to establish what that "reasonable" depth is, and set the control's limit

Reply:
Setting something like 1000 is not reasonable, but there could be scenarios where this could be desired. Linux has 650K commits and you may not be interested in the complete history.
In reality, very few will use depth at all, if you use it it will almost always be 1.

Reply:
There we go, we established an upper boundary 😄
So the updown control with the range [0, 1,000) and the default to 0 (that means 'full depth'). 
Correct?

Reply:
For Linux repo, 100K is a possible value. But we can leave that for the command line if you want a "neat" box.

Reply:
As much as I'd like GE to be the default git client for Linux devs, I think all of them choke themselves before using a Windows UI tool 
🤣

Reply:
Done :)

Reply:
> the gap is reserved for an existing Label, which I missed at first

It does look somewhat untidy. What is that label used for? 🤔